packages/meditrak-server/src/database/SurveyResponseChangeHandler/SurveyResponseChangeHandler.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import { OutdatedStatusUpdater } from './OutdatedStatusUpdater';
+
+const UPDATE_DEBOUNCE_TIME = 250; // wait 250ms after changes before updating, to avoid double-up
+
+export class SurveyResponseChangeHandler {
+  constructor(models, updateDebounceTime = UPDATE_DEBOUNCE_TIME) {
+    this.models = models;
+    this.updateDebounceTime = updateDebounceTime;
+    this.outdatedStatusUpdateCandidates = [];
+    this.scheduledUpdateTimeout = null;
+    this.scheduledUpdatePromise = null;
+    this.scheduledUpdatePromiseResolve = null;
+    this.activeUpdatePromise = null;
+    this.changeHandlerCancellers = [];
```

Comment:
I was heavily inspired (read: shamelessly copied) the batching logic currently defined in [AnalyticsRefresher.js](https://github.com/beyondessential/tupaia/blob/dev/packages/database/src/analytics/AnalyticsRefresher.js). This seems to be a useful pattern so perhaps worth abstracting under a shared class at some point!

I didn't feel it was worth the refactor in this PR, but if you feel strongly about it happy to give it a shot

Reply:
Also keen to break out the tricky part, and also happy for that to be a separate PR (though I'm not sure if a raised ticket would stand a good chance of making it into a sprint)